spec/unit/puppet/provider/powershell_spec.rb

```diff
@@ -0,0 +1,131 @@
+  let (:provider) { resource.provider }
+  let (:catalog) { Puppet::Resource::Catalog.new }
+
+  before :each do
+    #Puppet::FileSystem.stubs(:symlink)
+    resource.provider = provider
+  end
+
+  it "should be an instance of Puppet::Type::Base_dsc::ProviderPowershell" do
+    provider.must be_an_instance_of Puppet::Type::Base_dsc::ProviderPowershell
+  end
+
+  context ".set_symlink_folder" do
+    context "on Windows", :if => Puppet.features.microsoft_windows? do
+      it "should call Puppet::FileSystem.exist?" do
+        Puppet::FileSystem.stubs(:symlink)
+        Puppet::FileSystem.expects(:exist?).returns(true).times(1..2)
+
+        provider.set_symlink_folder
+      end
+
+      it "should not create a symlink if it already exists" do
+        Puppet::FileSystem.expects(:exist?).returns(true).times(1..2)
+        Puppet::FileSystem.expects(:symlink?).returns(true)
+        Puppet::FileSystem.expects(:readlink).returns(provider.vendored_modules_path)
+        Puppet::FileSystem.expects(:symlink).never
+
+        provider.set_symlink_folder
+      end
+
+      it "should recreate a symlink if the existing link path is wrong" do
+        Puppet::FileSystem.expects(:exist?).returns(true).times(1..2)
+        Puppet::FileSystem.expects(:symlink?).returns(true)
+        Puppet::FileSystem.expects(:readlink).returns('C:\dsc')
+        Puppet::FileSystem.expects(:symlink).once
+
+        provider.set_symlink_folder
+      end
+
+      it "should recreate a symlink if the existing path is not a symlink" do
+        Puppet::FileSystem.expects(:exist?).returns(true).times(1..2)
+        Puppet::FileSystem.expects(:symlink?).returns(false)
+        Puppet::FileSystem.expects(:symlink).once
+
+        provider.set_symlink_folder
+      end
+
+      it "should create a symlink when the symlink does not exist" do
+        Puppet::FileSystem.expects(:exist?).returns(false).times(1..2)
+        Puppet::FileSystem.expects(:symlink).once
+
+        provider.set_symlink_folder
+      end
+
+      it "should create the folder structure when it does not exist" do
+        Puppet::FileSystem.stubs(:symlink)
+
+        Puppet::FileSystem.expects(:exist?).returns(false).times(1..2)
+        Puppet::FileSystem.expects(:dir_exist?).returns(false)
+        Puppet::FileSystem.expects(:dir_mkpath).once
+
+        provider.set_symlink_folder
+      end
+    end
+
+    context "on Linux", :if => Puppet.features.posix? do
+      it "should not call Puppet::FileSystem.exist?" do
+        Puppet::FileSystem.expects(:exist?).never
+
+        provider.set_symlink_folder
+      end
+
+      it "should not attempt to create a symlink" do
+        Puppet::FileSystem.expects(:symlink).never
+
+        provider.set_symlink_folder
+      end
+    end
+  end
+
+  context ".get_program_files_dir", :if => Puppet.features.microsoft_windows? do
+
+    let (:reg_key) { mock('Win32_Registry') }
```

Comment:
Can't use this, and I can't use `let (:reg_key) { Win32::Registry}`. Both of them will register correctly, but mocks won't iterate over the evaluation block in a way that actually sets the `program_files_dir` variable.